src/routes/+page.svelte

```diff
@@ -58,27 +53,14 @@
 				content: ''
 			}
 		];
+		const data = await response.json();
+		let message = data;
 
-		const decoder = new TextDecoder('utf-8');
-
-		for await (const chunk of streamAsyncIterator(stream!)) {
-			console.log(decoder.decode(chunk));
-			history[history.length - 1].content += decoder.decode(chunk);
-
-			// TODO(max): Find a way to lock the actual completion from OpenAI, right now this will still use the full tokens
-			if (cancelStream) {
-				cancelStream = false;
-				break;
-			}
-		}
+		history[history.length - 1].content = message;
```

Comment:
We need to be doing streams here. I think it works pretty much the same as it does with normal langchain queries. But streams are really useful from a user perspective

Reply:
There is a option to stream it but for react it only stream the agent steps. So the Final answer will still only be one chunk. Streaming could be useful for logging though

Reply:
Taking a look at this I think it by default streams only the final output and not the intermediate steps. See https://js.langchain.com/docs/modules/model_io/chat/streaming. But if that doesn't work we can make a callback handler and have it create a stream.